src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const modifyState = { ...state };
+
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const entry of Object.entries(action.extraData)) {
+          modifyState[entry[0]] = entry[1];
+        }
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete modifyState[key];
+        }
+        break;
+      case 'clear':
+        for (const key of Object.keys(modifyState)) {
+          delete modifyState[key];
+        }
```

Comment:
To clear the 'modifyState', you don't need to loop through its keys to delete them one by one. You can simply assign an empty object to 'modifyState' to achieve the same result.